Fixes for running tests on Raspbian

JamesWTruher opened the pull request:
<!-- Anything that looks like this is a comment and can't be seen after the Pull Request is created. -->

# PR Summary

Fix 2 test files for running tests on Raspbian

<!-- Summarize your PR between here and the checklist. -->

## PR Context

<!-- Provide a little reasoning as to why this Pull Request helps and why you have opened it. -->

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

JamesWTruher committed:
Fix test to handle raspian error code for pinging bad host

JamesWTruher committed:
Skip MOF compilation tests on raspbian

PoshChan commented on test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:
```diff
@@ -64,11 +64,16 @@ Describe "Test-Connection" -tags "CI" {
             { $result = Test-Connection "fakeHost" -Count 1 -Quiet -ErrorAction Stop } |
```
@JamesWTruher, your last commit had 2 failures in `PowerShell-CI-linux`
DSC MOF Compilation.Should be able to compile a MOF from a basic configuration on Windows
```powershell
Exception calling "Create" with "1" argument(s): "At line:5 char:17
+                 File f1
+                 ~~~~
Undefined DSC resource 'File'. Use Import-DSCResource to import the resource."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PSDesiredStateConfiguration/MOF-Compilation.Tests.ps1: line 175
175:         [Scriptblock]::Create(@"

```
DSC MOF Compilation.Should be able to compile a MOF from a configuration with multiple resources on Windows
```powershell
Exception calling "Create" with "1" argument(s): "At line:5 char:17
+                 File f1
+                 ~~~~
Undefined DSC resource 'File'. Use Import-DSCResource to import the resource.

At line:10 char:17
+                 Script s1
+                 ~~~~~~
Undefined DSC resource 'Script'. Use Import-DSCResource to import the resource."
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PSDesiredStateConfiguration/MOF-Compilation.Tests.ps1: line 195
195:         [Scriptblock]::Create(@"

```

PoshChan replied:
@JamesWTruher, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.TraceRoute.Quiet works
```powershell
Expected $true, but got $null.
at <ScriptBlock>, /Users/runner/runners/2.164.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 272
272:             $result | Should -BeTrue
```

JamesWTruher committed:
change logic for skipping additional platforms

JamesWTruher committed:
Don't use hostname with -quiet test

There's no need to go after a specific host with -quiet, using localhost will prove the point. I've seen name lookup failures in CI (on Mac)

JamesWTruher commented:
@adityapatwardhan this looks ready to go

adityapatwardhan commented on test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:
```diff
@@ -64,11 +64,16 @@ Describe "Test-Connection" -tags "CI" {
-                $error[0].Exception.InnerException.ErrorCode | Should -Be -131073
-            } else {
-                $error[0].Exception.InnerException.ErrorCode | Should -Be 11001
+            if ((Get-PlatformInfo) -match "rasbian") {
```
```suggestion
            if ((Get-PlatformInfo) -match "raspbian") {
```

JamesWTruher committed:
Update test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1

Co-Authored-By: Aditya Patwardhan <adityap@microsoft.com>

adityapatwardhan commented:
@PoshChan please retry static

PoshChan commented:
@adityapatwardhan, successfully started retry of `PowerShell-CI-static-analysis`
